java/src/com/google/idea/blaze/java/fastbuild/FastBuildChangedFilesService.java

```diff
@@ -319,15 +323,25 @@ private ImmutableSet<File> getSourceFilesRecursively(
                     .map(decoder::decode)
                     .filter(f -> f.getName().endsWith(".java") || f.getName().endsWith(".scala"))
                     .forEach(sourceFiles::add);
+                addBuildFile(sourceFiles, workSpaceRootPath, d.buildFilePath());
               } else if (d.protoInfo().isPresent()) {
                 d.protoInfo().get().sources().stream()
                     .map(decoder::decode)
                     .forEach(sourceFiles::add);
+                addBuildFile(sourceFiles, workSpaceRootPath, d.buildFilePath());
               }
             });
     return ImmutableSet.copyOf(sourceFiles);
   }
 
+  private static void addBuildFile(Set<File> sourceFiles, String workSpaceRootPath,
+      String buildFilePath) {
+    File buildFile = new File(workSpaceRootPath, buildFilePath);
```

Comment:
Maybe use `Files` and `Path` here?

Reply:
Honest question, why would that be better? Or I don't understand what you mean

Reply:
Because Path is the more idiomatic tool when writing Java code.
Part of the challenge in Java is that you have all of the old APIs forever so the motivation to move to the newer better abstractions is low.